python/lsst/ts/externalscripts/maintel/make_comcam_calibrations.py

```diff
@@ -0,0 +1,131 @@
+# This file is part of ts_externalscripts
+#
+# Developed for the LSST Telescope and Site Systems.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+
+__all__ = ["MakeComCamCalibrations"]
+
+import yaml
+
+from lsst.ts.observatory.control.maintel.comcam import ComCam
+from ..base_make_calibrations import BaseMakeCalibrations
+
+
+class MakeComCamCalibrations(BaseMakeCalibrations):
+    """Class for taking images, constructing, verifying, and
+       certifying master calibrations with LSSTComCam.
+    """
+
+    def __init__(self, index=1):
+        super().__init__(
+            index=index,
+            descr="This class takes biases, darks, and flats with LSSTComCam, "
+                  "constructs a master calibration for each image type by calling "
+                  "the appropiate pipetask via OCPS, and then verifies and certifies "
+                  "each master calibration. "
+        )
+        self._comcam = ComCam(domain=self.domain, log=self.log)
+
+    @property
+    def camera(self):
+        return self._comcam
+
+    @property
+    def instrument_name(self):
+        """String with instrument name for pipeline task"""
+        return "LSSTComCam"
+
+    @property
+    def image_in_oods(self):
+        """Archiver"""
+        return self.camera.rem.ccarchiver.evt_imageInOODS
+
+    @classmethod
+    def get_schema(cls):
+        schema = """
+        $schema: http://json-schema.org/draft-07/schema#
+        $id: https://github.com/lsst-ts/ts_externalscripts/python/lsst/ts/\
+                externalscripts/maintel/make_comcam_calibrations.py
+        title: MakeComCamCalibrations v1
+        description: Configuration for making a LSSTComCam master calibrations SAL Script.
+        type: object
+        properties:
+            filter:
+                description: Filter name or ID; if omitted the filter is not changed.
+                anyOf:
+                  - type: string
+                  - type: integer
+                    minimun: 1
```

Comment:
Typo.

Can these schemas be derived from a common start (here and Latiss)?  This seems like a lot of code duplication, and if the only difference is the input collections and repo location (which differ only by a fixed string), then there should be a simpler way.

And I think this also answers my question of "where is the filter set", although I don't see where it's sent to the camera (I'm probably just missing something).

Reply:
I actually tried to move the redundant lines from the LSSTComCam and LATISS code to the schema in the base class. The only difference would be the instrument name. Unfortunately, there does not seem a way to access the methods in the base class (such as the ``instrument name`` method) from the class method ``get_schema``, so I'm going to have to revert to how things were.

Reply:
(to clarify, I tested the changes at the NTS and did not work; and consulted with Tiago to see if there was a way, but it doesn't look like it).